pydantic_settings/sources.py

```diff
@@ -674,6 +715,320 @@ def __repr__(self) -> str:
         )
 
 
+class CliSettingsSource(EnvSettingsSource):
```

Comment:
I haven't read all of this, I assume @hramezani has reviewed it?

Reply:
@hramezani please resolve this comment once satisfied.